src/features/runtime-checks.js

```diff
@@ -0,0 +1,249 @@
+            el[param] = this[param]
+        }
+
+        // Reflect all listeners to the new element
+        for (const [...args] of this.#listeners) {
+            el.addEventListener(...args)
+        }
+        this.#listeners = []
+
+        // Move all children to the new element
+        while (this.firstChild) {
+            el.appendChild(this.firstChild)
+        }
+
+        // Move the new element to the DOM
+        try {
+            this.insertAdjacentElement('afterend', el)
+        } catch (e) { console.warn(e) }
+
+        this.monitorProperties(el)
+        // TODO pollyfill WeakRef
+        this.#el = new WeakRef(el)
+
+        // Delay removal of the custom element so if the script calls removeChild it will still be in the DOM and not throw.
+        setTimeout(() => {
+            this.remove()
+        }, elementRemovalTimeout)
+    }
+
+    getElement () {
+        return this.#el?.deref()
+    }
+
+    setAttribute (name, value) {
+        const el = this.getElement()
+        if (el) {
+            return el.setAttribute(name, value)
+        }
+        return super.setAttribute(name, value)
+    }
+
+    removeAttribute (name) {
+        const el = this.getElement()
+        if (el) {
+            return el.removeAttribute(name)
+        }
+        return super.removeAttribute(name)
+    }
+
+    addEventListener (...args) {
+        const el = this.getElement()
+        if (el) {
+            return el.addEventListener(...args)
+        }
+        this.#listeners.push([...args])
+    }
+
+    removeEventListener (...args) {
+        const el = this.getElement()
+        if (el) {
+            return el.removeEventListener(...args)
+        }
+        this.#listeners = this.#listeners.filter((listener) => {
+            return listener[0] !== args[0] || listener[1] !== args[1]
+        })
+    }
+
+    toString () {
+        // TODO
+        let interfaceName = ''
+        if (this._tagName === 'script') {
+            interfaceName = 'Script'
+        }
+        return `[object HTML${interfaceName}Element]`
+    }
+
+    get tagName () {
+        return this.#tagName.toUpperCase()
+    }
+
+    get nodeName () {
+        return this.tagName
+    }
+
+    remove () {
+        const el = this.getElement()
+        if (el) {
+            return el.remove()
+        }
+        return super.remove()
+    }
+
+    removeChild (child) {
+        const el = this.getElement()
+        if (el) {
+            return el.removeChild(child)
+        }
+        return super.removeChild(child)
+    }
+}
+
+/**
+ * Overrides the instanceof checks to make the custom element interface pass an instanceof check
+ * @param {Object} elementInterface
+ */
+function overloadInstanceOfChecks (elementInterface) {
+    const proxy = new Proxy(elementInterface[Symbol.hasInstance], {
+        apply (fn, scope, args) {
+            if (args[0] instanceof DDGRuntimeChecks) {
+                return true
+            }
```

Comment:
Does this make `x instanceof A` always true if x is a `DDGRuntimeChecks`? Why do we need to do that?

Reply:
so that ddg-runtime-check behaves like script.

This means if the page checks this; it pretends to be a script (same as tagname etc) so they should continue interacting with it.